### PR TITLE
Enable System.Collections and ResourceManager for NXP target

### PR DESCRIPTION
**Pull request by MateuszKlatecki**
## How Has This Been Tested?

Tested with samples: Collections ManagedResources

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: Mateusz Klatecki <mateusz.klatecki@gc5.pl>

### CODE REVIEW

File: cmake-variants.TEMPLATE-MIMXRT1060.json [hunk 1]
```diff
@@ -36,23 +36,25 @@
           "SUPPORT_ANY_BASE_CONVERSION": "ON",
           "RTOS": "FREERTOS",
           "FREERTOS_BOARD": "NXP_MIMXRT1060_EVK",
-          "FREERTOS_VERSION": "v1.4.8",
+          "FREERTOS_VERSION": "V10.2.1",
```
Comment (josesimoes) [inline cmake-variants.TEMPLATE-MIMXRT1060.json]: This change doesn't belong here...

File: cmake-variants.TEMPLATE-MIMXRT1060.json [hunk 2]
```diff
@@ -36,23 +36,25 @@
           "SUPPORT_ANY_BASE_CONVERSION": "ON",
           "RTOS": "FREERTOS",
           "FREERTOS_BOARD": "NXP_MIMXRT1060_EVK",
-          "FREERTOS_VERSION": "v1.4.8",
+          "FREERTOS_VERSION": "V10.2.1",
           "CMSIS_VERSION": "5.5.1",
-          "FATFS_VERSION": "R0.13c",
+          "FATFS_VERSION": "R0.14",
```
Comment (josesimoes) [inline cmake-variants.TEMPLATE-MIMXRT1060.json]: This change doesn't belong here...

File: cmake-variants.TEMPLATE-MIMXRT1060.json [hunk 3]
```diff
@@ -36,23 +36,25 @@
...
-          "FATFS_VERSION": "R0.13c",
+          "FATFS_VERSION": "R0.14",
           "LWIP_VERSION": "STABLE-2_0_3_RELEASE",
           "NF_BUILD_RTM": "OFF",
           "NF_FEATURE_DEBUGGER": "ON",
           "NF_FEATURE_HAS_SDCARD": "ON",
           "NF_FEATURE_RTC": "ON",
           "NF_FEATURE_HAS_CONFIG_BLOCK": "ON",
-          "NF_SECURITY_OPENSSL": "OFF",
+          "NF_SECURITY_MBEDTLS": "OFF",
```
Comment (josesimoes) [inline cmake-variants.TEMPLATE-MIMXRT1060.json]: This change doesn't belong here...

**Comment by MateuszKlatecki** [comment]
Fixed

**Pull request merged**